server/plugin.go

```diff
@@ -29,41 +29,69 @@ func (p *Plugin) ServeHTTP(c *plugin.Context, w http.ResponseWriter, r *http.Req
 	if !config.Enabled || config.Secret == "" || config.UserName == "" {
 		http.Error(w, "This plugin is not configured.", http.StatusForbidden)
 		return
-	} else if r.URL.Path != "/webhook" {
+	}
+
+	if r.URL.Path != "/webhook" {
 		http.NotFound(w, r)
 		return
-	} else if r.Method != http.MethodPost {
-		http.Error(w, "method not allowed", http.StatusMethodNotAllowed)
+	}
+
+	if r.Method != http.MethodPost {
+		http.Error(w, "Method not allowed.", http.StatusMethodNotAllowed)
 		return
-	} else if subtle.ConstantTimeCompare([]byte(r.URL.Query().Get("secret")), []byte(config.Secret)) != 1 {
+	}
+
+	if subtle.ConstantTimeCompare([]byte(r.URL.Query().Get("secret")), []byte(config.Secret)) != 1 {
 		http.Error(w, "You must provide the configured secret.", http.StatusForbidden)
 		return
 	}
 
-	var webhook Webhook
+	channelID := r.URL.Query().Get("channel")
+	if channelID == "" {
+		http.Error(w, "You must provide a channelID.", http.StatusBadRequest)
+		return
+	}
+	teamName := r.URL.Query().Get("team")
+	// Can be "" for DM
```

Comment:
I'm not sure on this one. I just copied the old behaviour. Does someone know, if you can call `GetChannelByNameForTeamName()` with an empty teamName to get DM's?

Reply:
It doesn't look like it. That API seems to require a team https://github.com/mattermost/mattermost-server/blob/master/app/channel.go#L1085. If you give a real team name and the channel name of a DM (concatenated user IDs) you could get a DM but it's hardly a normal use case

Reply:
Okay, changed to code to require a team name.